tests/fsharp/Compiler/ErrorMessages/SuggestionsTests.fs

```diff
@@ -0,0 +1,320 @@
+
+let x = N.``longe name``
+            """
+            FSharpErrorSeverity.Error
+            39
+            (5, 11, 5, 25)
+            "The value, constructor, namespace or type 'longe name' is not defined. Maybe you want one of the following:\r\n   longer name"
+
+
+    [<Test>]
+    let ``Suggest Double Backtick Unions`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module N =
+    type MyUnion =
+    | ``My Case1``
+    | Case2
+
+open N
+
+let x = N.MyUnion.``My Case2``
+            """
+            FSharpErrorSeverity.Error
+            39
+            (9, 19, 9,31)
+            "The field, constructor or member 'My Case2' is not defined. Maybe you want one of the following:\r\n   My Case1\r\n   Case2"
+
+
+    [<Test>]
+    let ``Suggest Fields In Constructor`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+type MyClass() =
+    member val MyProperty = "" with get, set
+    member val MyProperty2 = "" with get, set
+    member val ABigProperty = "" with get, set
+
+let c = MyClass(Property = "")
+            """
+            FSharpErrorSeverity.Error
+            495
+            (7, 17, 7, 25)
+            "The object constructor 'MyClass' has no argument or settable return property 'Property'. The required signature is new : unit -> MyClass. Maybe you want one of the following:\r\n   MyProperty\r\n   MyProperty2\r\n   ABigProperty"
+
+
+    [<Test>]
+    let ``Suggest Generic Type`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+type T = System.Collections.Generic.Dictionary<int11,int>
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 48, 2, 53)
+            "The type 'int11' is not defined. Maybe you want one of the following:\r\n   int16\r\n   int16`1\r\n   int8\r\n   uint16\r\n   int"
+
+
+    [<Test>]
+    let ``Suggest Methods`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Test2 =
+    type D() =
+
+       static let x = 1
+
+       member x.Method1() = 10
+
+    D.Method2()
+            """
+            FSharpErrorSeverity.Error
+            39
+            (9, 7, 9, 14)
+            "The field, constructor or member 'Method2' is not defined. Maybe you want one of the following:\r\n   Method1"
+
+
+    [<Test>]
+    let ``Suggest Modules`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+module Collections =
+
+    let f () = printfn "%s" "Hello"
+
+open Collectons
+            """
+            FSharpErrorSeverity.Error
+            39
+            (6, 6, 6, 16)
+            "The namespace or module 'Collectons' is not defined. Maybe you want one of the following:\r\n   Collections"
+
+
+    [<Test>]
+    let ``Suggest Namespaces`` () =
+        CompilerAssert.TypeCheckSingleError
+            """
+open System.Collectons
+            """
+            FSharpErrorSeverity.Error
+            39
+            (2, 13, 2, 23)
+            "The namespace 'Collectons' is not defined."
```

Comment:
No suggestion here? Expected to see a Collections suggestion

Reply:
need to revisit. but not part of this PR